src/remote-config/remote-config-api-client.ts

```diff
@@ -106,12 +106,16 @@ export class RemoteConfigApiClient {
     this.httpClient = new AuthorizedHttpClient(app);
   }
 
-  public getTemplate(): Promise<RemoteConfigTemplate> {
+  public getTemplate(versionNumber?: number | string): Promise<RemoteConfigTemplate> {
+    let path = 'remoteConfig';
+    if (versionNumber) {
```

Comment:
If the `versionNumber` is an invalid string (NaN)the backend API will return an error similar to the following:
```
errorInfo:
   { code: 'remote-config/invalid-argument',
     message: 'Invalid value at \'version_number\' (TYPE_INT64), "abc"' },
  codePrefix: 'remote-config' }
```

We could add a check here before making the request, but converting a `string` to a `number` to check `isNaN` will be tricky as the `string` is in `int64 format`. Appreciate any thoughts on this! :)

Reply:
We could also run a RegEx to check if the `string` only contains numbers @hiranya911

Reply:
How about:

```
const version = validators.isNumber(versionNumber) ? versionNumber : parseFloat(versionNumber);
if (!Number.isInteger(version)) {
  throw
}
```

Reply:
Thanks! I am fine with this. If the string prefixes with numbers and followed by characters, `parseFloat` ignores the characters and parses the number in string.

ex: `70abc` becomes `70`

I believe this is okay, unless we would like further validate with a regex